Failed to publish Java SDK

```
FAILURE: Build failed with an exception.

* What went wrong:
Task 'publishToSonatype' not found in root project 'com.pulumi.command'.
```

https://github.com/pulumi/pulumi-command/actions/runs/3397047954/jobs/5649020568

--- a/provider/cmd/pulumi-resource-command/schema.json
+++ b/provider/cmd/pulumi-resource-command/schema.json
@@ -448,6 +448,7 @@
     "language": {
         "java": {
             "buildFiles": "gradle",
+            "gradleNexusPublishPluginVersion": "1.1.0",
             "dependencies": {
                 "com.pulumi:pulumi": "0.6.0",
                 "com.google.code.gson:gson": "2.8.9",

--- a/sdk/java/build.gradle
+++ b/sdk/java/build.gradle
@@ -5,6 +5,7 @@ plugins {
     id("signing")
     id("java-library")
     id("maven-publish")
+    id("io.github.gradle-nexus.publish-plugin") version "1.1.0"
 }
 
 group = "com.pulumi"
@@ -16,7 +17,7 @@ def resolvedVersion = System.getenv("PACKAGE_VERSION") ?:
 
 def signingKey = System.getenv("SIGNING_KEY")
 def signingPassword = System.getenv("SIGNING_PASSWORD")
-def publishRepoURL = System.getenv("PUBLISH_REPO_URL")
+def publishRepoURL = System.getenv("PUBLISH_REPO_URL") ?: "https://s01.oss.sonatype.org"
 def publishRepoUsername = System.getenv("PUBLISH_REPO_USERNAME")
 def publishRepoPassword = System.getenv("PUBLISH_REPO_PASSWORD")
 
@@ -116,19 +117,6 @@ publishing {
             }
         }
     }
-
-    if (publishRepoURL) {
-        repositories {
-            maven {
-                name = "PublishRepo"
-                url = publishRepoURL
-                credentials {
-                    username = publishRepoUsername
-                    password = publishRepoPassword
-                }
-            }
-        }
-    }
 }
 
 javadoc {
@@ -136,6 +124,19 @@ javadoc {
         options.addBooleanOption('html5', true)
     }
     options.jFlags("-Xmx2g", "-Xms512m")
+}
+
+if (publishRepoUsername) {
+    nexusPublishing {
+        repositories {
+            sonatype {
+                nexusUrl.set(uri(publishRepoURL + "/service/local/"))
+                snapshotRepositoryUrl.set(uri(publishRepoURL + "/content/repositories/snapshots/"))
+                username = publishRepoUsername
+                password = publishRepoPassword
+            }
+        }
+    }
 }
 
 if (signingKey) {

